usaspending_api/search/v2/views/spending_by_award.py

```diff
@@ -287,6 +278,30 @@ def query_elasticsearch(self) -> list:
         filter_query = QueryWithFilters.generate_awards_elasticsearch_query(self.filters)
         sort_field = self.get_elastic_sort_by_fields()
         sorts = [{field: self.pagination["sort_order"]} for field in sort_field]
+        record_num = (self.pagination["page"] - 1) * self.pagination["limit"]
+        if record_num >= settings.ES_AWARDS_MAX_RESULT_WINDOW and not self.last_id and not self.last_value:
+            sorts.append({"award_id": self.pagination["sort_order"]})
+            logger.warning(
+                "Jumping to random page past 50,000 records will have a performance hit when using Elasticsearch"
+            )
+            self.pagination["upper_bound"] = record_num
+            self.pagination["lower_bound"] = record_num - 1
+            queryset = self.construct_queryset()
```

Comment:
I see now. So this is where it runs a database query for 1 supposed record, in order to use that db record to jump the search_after ahead. 

It's not elegant, and does chain us back to the db. But it's probably more efficient than what I was thinking - which was to "scroll" through the ES pages one page at a time on the server-side until you reached the desired page.

This does run the risk of not accurately paging to the right page in ES if the source DB data (matview) and ES index are not completely in sync.